datacapturing/src/main/java/de/cyface/datacapturing/backend/CapturingProcess.kt

```diff
@@ -48,60 +44,48 @@ import kotlin.math.abs
 import kotlin.math.min
 
 /**
- * Implements the data capturing functionality for Cyface. This class implements the SensorEventListener to listen to
- * acceleration sensor events as well as the LocationListener to listen to location updates.
+ * Implements the data capturing functionality for Cyface.
+ *
+ * This class implements the SensorEventListener to listen to acceleration sensor events as well as
+ * the LocationListener to listen to location updates. - FIXME: Sensor event listener should be optional
  *
  * @author Klemens Muthmann
  * @author Armin Schnabel
  * @version 3.2.2
  * @since 1.0.0
+ * @param locationCapture The [LocationCapture] which sets up the location capturing.
+ * @param sensorCapture The [SensorCapture] implementation which decides if sensor data should
+ * be captured and sets up the capturing if requested.
+ * @throws `SecurityException` If user did not provide permission to access geo location.
  */
-@SuppressLint("MissingPermission") // UI has to handle permission request before starting this
 abstract class CapturingProcess internal constructor(
-    locationManager: LocationManager, sensorService: SensorManager,
-    geoLocationDeviceStatusHandler: GeoLocationDeviceStatusHandler,
-    locationEventHandlerThread: HandlerThread,
-    sensorEventHandlerThread: HandlerThread, sensorFrequency: Int
+    private val locationCapture: LocationCapture,
+    private val sensorCapture: SensorCapture,
```

Comment:
Since these are classes they should be called "Capturer", which is the noun. Capture is the verb.

Reply:
`Capture` is a noun, too. It's basically a style-thing. E.g. Android's [ImageCapture](https://developer.android.com/reference/androidx/camera/core/ImageCapture) class.

I went with `Capture` instead of `Capturer` as the later is a bit harder to read/pronounce.